manager/controllers/packages.go

```diff
@@ -63,18 +85,40 @@ func PackagesListHandler(c *gin.Context) {
 
 	query := packagesQuery(account)
 	query, meta, links, err := ListCommon(query, c, "/packages", PackagesOpts)
+	query = ApplySearch(c, query, "pn.name")
 	query, _ = ApplyTagsFilter(c, query, "sp.inventory_id")
 
 	if err != nil {
 		LogAndRespError(c, err, "database error")
 		return
 	}
-	var systems []PackageItem
-	err = query.Scan(&systems).Error
+	var itemQ []PackageItemQuery
+	err = query.Find(&itemQ).Error
 	if err != nil {
 		LogAndRespError(c, err, "database error")
 		return
 	}
+	nameIds := make([]int, len(itemQ))
+	for i, s := range itemQ {
+		nameIds[i] = s.NameID
+	}
+
+	var summaries []packageSummary
+	summaryQuery := packageDescQuery(nameIds)
```

Comment:
I would rather to do it using single Db query, as it would be more robust solution. What exactly is the reason you do it this way? Could you add more comments. It's quite complex code.

Reply:
I'd rather use single query as well, but it got too complicated. By separating it into 2 queries we were able to get better performance.